android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboardJSHandler.java

```diff
@@ -284,7 +285,17 @@ private static void performLeftDeletions(InputConnection ic, int dn) {
 
     // Chop dn+numPairs code points from the end of charsBackup
     // subSequence indices are start(inclusive) to end(exclusive)
-    CharSequence expectedChars = charsBackup.subSequence(0, charsBackup.length() - (dn + numPairs));
+    int start = 0;
+    int end = charsBackup.length() - (dn + numPairs);
+    CharSequence expectedChars;
+    try {
+      expectedChars = charsBackup.subSequence(start, end);
+    } catch (IndexOutOfBoundsException e) {
+      KMLog.LogException(TAG,
```

Comment:
Good news is @jahorton 's found the root cause for the original issue.

We still probably need to keep these catch blocks to prevent future crashes. Just don't log to avoid messy toast notifications?

Reply:
That fix: https://github.com/keymanapp/keyman/pull/10869

To be clear, it fixes the markedly increased frequency with which we see the bug in 17.0.  It probably also fixes the issue title outright.  But... I can't promise that it fixes **_all_** related Sentry errors; the fix shouldn't even touch whatever caused the issue in 16.0 and before.

Reply:
Checking a 16.0 error report... it looks like someone used a backspace on an empty context.  That can totally happen now, too, but that case - "empty context" - does have a null guard already, which is L279-L280, and it existed back then too.  Hmm.